app/src/main/java/com/example/fitjournal/home/presentation/components/card/CalisthenicsCard.kt

```diff
@@ -60,35 +66,52 @@ fun CalisthenicsCard(
 }
 
 @Composable
-private fun MostRecentSet(
-    reps: Int?,
-    time: String?,
+private fun MostRecentCalisthenicsSet(
+    calisthenicsUi: CalisthenicsUi,
     modifier: Modifier = Modifier
 ) {
     Column(
         modifier = modifier,
         verticalArrangement = Arrangement.SpaceBetween,
         horizontalAlignment = Alignment.Start
     ) {
-        reps?.let {
+        if (calisthenicsUi.sets != null && calisthenicsUi.reps != null) {
             Text(
```

Comment:
I left a comment about moving null checks to an extension function inside data class, we can do that here